### PR TITLE
Rename CUSTOM_PME_PARAMETERS to ALIGNMENT_PARAMETERS, add endpoint fo…

### PR DESCRIPTION
**Pull request by dominikbrazdil**
…r retrieving default alignment parameters for build types

### Checklist:

* [x] Have you added a note in the [CHANGELOG wiki](https://github.com/project-ncl/pnc/wiki/Changelog) for your change if user-facing?
* [x] Have you added unit tests for your change?

### CODE REVIEW

File: moduleconfig/src/main/java/org/jboss/pnc/common/json/moduleconfig/AlignmentConfig.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+public class AlignmentConfig extends AbstractModuleConfig {
+
+    private final static Logger log = LoggerFactory.getLogger(AlignmentConfig.class);
+
+//    private List<String> mavenParameters;
+//
+//    private List<String> npmParameters;
+//
+//    private List<String> gradleParameters;
+
```
Comment (jbartece) [inline moduleconfig/src/main/java/org/jboss/pnc/common/json/moduleconfig/AlignmentConfig.java]: Clean the code

File: moduleconfig/src/main/java/org/jboss/pnc/common/json/moduleconfig/AlignmentConfig.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    private Map<String, List<String>> alignmentParameters;
+
+    public AlignmentConfig(
+//            @JsonProperty("mavenParameters") List<String> mavenParameters,
+//            @JsonProperty("npmParameters") List<String> npmParameters,
+//            @JsonProperty("gradleParameters") List<String> gradleParameters
+            @JsonProperty("alignmentParameters") Map<String, List<String>> alignmentParameters) {
+//        this.mavenParameters = mavenParameters;
+//        this.npmParameters = npmParameters;
+//        this.gradleParameters = gradleParameters;
```
Comment (jbartece) [inline moduleconfig/src/main/java/org/jboss/pnc/common/json/moduleconfig/AlignmentConfig.java]: Same here.

File: moduleconfig/src/main/java/org/jboss/pnc/common/json/moduleconfig/AlignmentConfig.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+    private final static Logger log = LoggerFactory.getLogger(AlignmentConfig.class);
+
+//    private List<String> mavenParameters;
+//
+//    private List<String> npmParameters;
+//
+//    private List<String> gradleParameters;
+
+    private Map<String, List<String>> alignmentParameters;
```
Comment (jbartece) [inline moduleconfig/src/main/java/org/jboss/pnc/common/json/moduleconfig/AlignmentConfig.java]: Document the expected values in the Map, please.

File: moduleconfig/src/main/java/org/jboss/pnc/common/json/moduleconfig/AlignmentConfig.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+    public AlignmentConfig(
+//            @JsonProperty("mavenParameters") List<String> mavenParameters,
+//            @JsonProperty("npmParameters") List<String> npmParameters,
+//            @JsonProperty("gradleParameters") List<String> gradleParameters
+            @JsonProperty("alignmentParameters") Map<String, List<String>> alignmentParameters) {
```
Comment (jbartece) [inline moduleconfig/src/main/java/org/jboss/pnc/common/json/moduleconfig/AlignmentConfig.java]: We store the generic parameters as a single String in the DB and also will provide it as one String in the UI.
I think is should be just a Map<String, String>.

File: rest-api/src/main/java/org/jboss/pnc/rest/api/endpoints/BuildConfigurationEndpoint.java [hunk 5]
```diff
@@ -380,4 +378,16 @@ BuildConfiguration restoreRevision(
     @Path("/supported-parameters")
     Set<org.jboss.pnc.dto.response.Parameter> getSupportedParameters();
 
+    @Operation(summary = "Provides list of default alignment parameters according to build type.",
+            description = "Provides default parameters for build configuration set in resource config file.",
```
Comment (jbartece) [inline rest-api/src/main/java/org/jboss/pnc/rest/api/endpoints/BuildConfigurationEndpoint.java]: "set in resource config file" - it is implementation detail, which shouldn't be on the REST API. The important information is, that these strings are the default params.

File: rest-api/src/main/java/org/jboss/pnc/rest/api/endpoints/BuildConfigurationEndpoint.java [hunk 6]
```diff
@@ -380,4 +378,16 @@ BuildConfiguration restoreRevision(
     @Path("/supported-parameters")
     Set<org.jboss.pnc.dto.response.Parameter> getSupportedParameters();
 
+    @Operation(summary = "Provides string of default alignment parameters.",
+            description = "Provides default parameters for build configuration according to build type chosen by user.",
```
Comment (jbartece) [inline rest-api/src/main/java/org/jboss/pnc/rest/api/endpoints/BuildConfigurationEndpoint.java]: I think it should be: according to a build type chosen by a user

File: facade/src/main/resources/buildConfigurationSupportedGenericParameters.properties [hunk 7]
```diff
@@ -16,7 +16,7 @@
 # limitations under the License.
 #
 
-CUSTOM_PME_PARAMETERS=Additional parameters, which will be passed to the PME CLI executable during the alignment before the build. The format is as you would enter them on a command line, and each MUST start with a dash.
+ALIGNMENT_PARAMETERS=Additional parameters, which will be passed to the PME CLI executable during the alignment before the build. The format is as you would enter them on a command line, and each MUST start with a dash.
 BREW_BUILD_NAME=Specify the Brew build name of the build configuration. This is used to override the default value, and can be useful for builds that disable PME. For Maven builds the format should be '<groupid>:<artifactid>'.
 CUSTOM_PROJECT_MANIPULATOR_PARAMETERS=Additional parameters, which will be passed to the Project Manipulator CLI executable during the alignment (NPM, etc, non-Maven projects) before the build. The format is as you would enter them on a command line, and each MUST start with a dash.
```
Comment (pkocandr) [inline facade/src/main/resources/buildConfigurationSupportedGenericParameters.properties]: CUSTOM_PROJECT_MANIPULATOR_PARAMETERS shopuld have been removed as ALIGNMENT_PARAMETERS replaces it, right?

Comment (jbartece) [reply facade/src/main/resources/buildConfigurationSupportedGenericParameters.properties]: Correct. I missed. Thanks @pkocandr 
@dominikbrazdil Please create a new PR to fix it.

**Pull request merged**